drivers/regulator/regulator_shell.c

```diff
@@ -109,6 +109,98 @@ static int cmd_set_ilim(const struct shell *sh, size_t argc, char **argv)
 	return 0;
 }
 
+static int cmd_set_mode(const struct shell *sh, size_t argc, char **argv)
+{
+	int mode, ret;
+	const struct device *reg_dev;
+
+	reg_dev = device_get_binding(argv[1]);
+	if (!device_is_ready(reg_dev)) {
```

Comment:
I guess you can do a simple NULL check just because `device_get_binding` has a check to see if the object was successfully initialized on startup.

Reply:
Thanks for the suggestion, I've switched to a `NULL` check. Also cleaned up usages of `device_get_binding` in the remainder of this shell module